src/CmsBundle/Repository/ContentRepository.php

```diff
@@ -165,16 +164,43 @@ public function findActiveAddressable()
      */
     public function getRelatedContentToBlocks($search)
     {
-        return $this->createQueryBuilder('c')
-                ->select('c')
-                ->innerjoin('c.blocks', 'b', 'WITH', 'c.id = b.content')
-                ->where('b.value LIKE :search')
-                ->orWhere('c.title LIKE :search')
-                ->orWhere('c.description LIKE :search')
-                ->setParameter('search', '%'.$search.'%')
-                ->groupBy('c.id')
-                ->orderBy('c.id')
-                ->getQuery()
-                ->getResult();
+        $results = $this->createQueryBuilder('c')
+                        ->select('c')
+                        ->innerjoin('c.blocks', 'b', 'WITH', 'c.id = b.content')
+                        ->where('b.value LIKE :search')
+                        ->orWhere('c.title LIKE :search')
+                        ->orWhere('c.description LIKE :search')
+                        ->setParameter('search', '%'.$search.'%')
+                        ->groupBy('c.id')
+                        ->orderBy('c.id')
+                        ->getQuery()
+                        ->getResult();
+
+        return $this->sortSearchResults($results, $search);
+    }
+
+    /**
+     * Sort search results by giving priority to founded by title.
+     *
+     * @param array  $results
+     * @param string $search
+     *
+     * @return ArrayCollection
+     */
+    public function sortSearchResults($results, $search)
+    {
+        $sortedResults = array();
+
+        if (!empty($results)) {
+            foreach ($results as $result) {
+                if (stripos($result->getTitle(), $search)) {
+                    array_unshift($sortedResults, $result);
+                } else {
+                    $sortedResults[] = $result;
```

Comment:
This would mean that when a result does not have the match in the title, it will empty the whole `sortedResults` array, right?
I guess you need to remove the `else`.